conan/tools/meson/meson.py

```diff
@@ -14,24 +13,26 @@ def __init__(self, conanfile, build_folder='build'):
     def _build_dir(self):
         build = self._conanfile.build_folder
         if self._build_folder:
-            build = os.path.join(self._conanfile.build_folder, self._build_folder)
+            build = join(self._conanfile.build_folder, self._build_folder)
         return build
 
     def configure(self, source_folder=None):
         source = self._conanfile.source_folder
         if source_folder:
-            source = os.path.join(self._conanfile.source_folder, source_folder)
+            source = join(self._conanfile.source_folder, source_folder)
 
         cmd = "meson setup"
+        gen_folder = self._conanfile.generators_folder
         if cross_building(self._conanfile):
-            cmd += ' --cross-file "{}"'.format(MesonToolchain.cross_filename)
+            cmd += ' --cross-file "{}"'.format(join(gen_folder, MesonToolchain.cross_filename))
         else:
-            cmd += ' --native-file "{}"'.format(MesonToolchain.native_filename)
+            cmd += ' --native-file "{}"'.format(join(gen_folder, MesonToolchain.native_filename))
         cmd += ' "{}" "{}"'.format(self._build_dir, source)
         if self._conanfile.package_folder:
             cmd += ' -Dprefix="{}"'.format(self._conanfile.package_folder)
+        vcvars = join(gen_folder, "conanvcvars")
```

Comment:
This seems to be added un-conditionally, and most likely it should only happen in some cases ("Visual" or "msvc" compiler). Furthermore, this doesn't need to be added explicitly, as long as the ``VCVars`` generator was called, it added the conanvcvars script to the "build" scripts, and it will be automatically invoked by ``self.run()``

Reply:
I will take another look into it

Reply:
Vcvars has been deleted from this file at all, probably because of the above reason. I will leave it as it was.